backend/internal/controllers/book.go

```diff
@@ -51,8 +53,29 @@ func (c *BookController) GetBookDetails(rw http.ResponseWriter, req *http.Reques
 
 // Creates a book and inserts it into the book database (/books)
 func (c *BookController) CreateBook(rw http.ResponseWriter, req *http.Request) {
+
+    // Check auth token
+    uid, ok := req.Context().Value("user").(string)
+	if !ok {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		fmt.Println("unable to get user from request context")
+		return
+	}
+
+    // Check admin permissions
+    isAdmin, _, canUploadBooks, _, err := c.Admins.FetchAdminPermissions(req.Context(), uid)
+    if err != nil {
+        writeResponse(rw, http.StatusInternalServerError, "error")
+        fmt.Println(err)
+        return
+    }
+    if !isAdmin || !canUploadBooks {
+        writeResponse(rw, http.StatusForbidden, "do not have permission")
+        return
+    }
```

Comment:
I would extract all of this into middleware - let's make some kind of a `RequireAdmin()` or `RequirePermission(CanManageUsers)` middleware in internal/controllers/middleware, then add it to the POST/PATCH/DELETE book routes.